src/shogun/lib/any.h

```diff
@@ -396,28 +393,19 @@ namespace shogun
 			return policy->type_info();
 		}
 
-	private:
-		void assert_same_policy_type(BaseAnyPolicy* other_policy)
```

Comment:
I understand we don't need this anymore? Is that why it is removed?

Reply:
Yes, but we need another check now